packages/calcite-components/src/components/combobox/combobox.tsx

```diff
@@ -281,6 +283,8 @@ export class Combobox
 
   @state() selectedVisibleChipsCount = 0;
 
+  @state() selectAll: boolean = false;
```

Comment:
Is there an e2e test for when all are selected upon initialization and the allSelected checkbox is checked?

Reply:
Can we rename this to `allSelected`? As it is it sounds like a method and not a state.

Reply:
There should be one source of truth for whether all items are selected. The source of truth should be if all the items are actually selected, not whether the select all item is checked. That should just reflect the state of all items being selected.

Reply:
Ideally, we would have something like this:

```tsx
@state()
private items: HTMLCalciteComboboxItemElement["el"][] = [];

@state()
private selectedItems: HTMLCalciteComboboxItemElement["el"][] = [];

@state()
get itemLength = (): number => {
  return this.items.length;
}

@state()
get selectedItemLength = (): number => {
  return this.selectedItems.length
}

@state()
get isAllSelected = (): boolean => {
  return this.selectedItemLength === this.itemLength;
}
```

Reply:
Then we can just use those stateful props to render accordingly. Right now, there's too many calls to `isAllSelected()` which won't trigger a re-render if selected items change.

Reply:
> Is there an e2e test for when all are selected upon initialization and the allSelected checkbox is checked?

Yep. https://github.com/Esri/calcite-design-system/pull/11721/files#diff-85da4d8c5247e61a3626fd500c5027b510000a530ba6972f87e27a2acf71a033R3137-R3149

Reply:
> Can we rename this to `allSelected`? As it is it sounds like a method and not a state.

I called it `selectAll` because it refers to `selectAllEnabled` combobox item element in the code. `allSelected` would mean that checkbox is checked.